clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -1287,6 +1287,16 @@ public void testCommitAsyncNegativeOffset() {
         assertTrue(mockOffsetCommitCallback.exception instanceof IllegalArgumentException);
     }
 
+    @Test(expected = CommitFailedException.class)
+    public void testCommitOffsetSyncCallbackWithoutNonRetriableException() {
```

Comment:
Ah, I see that now after quickly running the test without fix. Thanks for the clear explanation @ijuma. However I am not sure how to create the condition which causes the future not to complete from `client.poll` method. Can you please add some pointers for this?